include/gamepad/controller.hpp

```diff
@@ -17,31 +18,152 @@ enum EventType {
     ON_PRESS,
     ON_LONG_PRESS,
     ON_RELEASE,
+    ON_SHORT_RELEASE,
 };
 
 class Button {
         friend class Controller;
     public:
+        /// Whether the button has just been pressed
         bool rising_edge = false;
+        /// Whether the button has just been released
         bool falling_edge = false;
+        /// Whether the button is currently held down
         bool is_pressed = false;
-        uint32_t last_press_time = pros::millis();
-        uint32_t last_release_time = last_press_time;
+        /// How long the button has been held down
         uint32_t time_held = 0;
+        /// How long the button has been released
         uint32_t time_released = 0;
+        /// How long the threshold should be for the longPress and shortRelease events
         uint32_t long_press_threshold = 500;
+        /**
+         * @brief Register a function to run when the button is pressed.
+         *
+         * @param listenerName The name of the listener, this must be a unique name
+         * @param func The function to run when the button is pressed, the function MUST NOT block
+         * @return true The listener was successfully registered
+         * @return false The listener was not successfully registered (there is already a listener with this name)
+         * Example:
+         * @code {.cpp}
+         *   // Use a function...
+         *   Gamepad::master.Down.onPress("downPress1", downPress1);
+         *   // ...or a lambda
+         *   Gamepad::master.Up.onPress("upPress1", []() { std::cout << "I was pressed!" << std::endl; });
+         * @endcode
+         *
+         */
+        bool onPress(std::string listenerName, std::function<void(void)> func) const;
+        /**
+         * @brief Register a function to run when the button is long pressed. WARNING: When using this event along
+         * with onPress, both the onPress and onlongPress listeners may fire together. (The button has been held
+         * down for 500ms or more, this threshold can be adjusted by changing long_press_threshold).
+         *
+         * @param listenerName The name of the listener, this must be a unique name
+         * @param func The function to run when the button is long pressed, the function MUST NOT block
+         * @return true The listener was successfully registered
+         * @return false The listener was not successfully registered (there is already a listener with this name)
```

Comment:
pad the returns with the example code

```suggestion
         * @return false The listener was not successfully registered (there is already a listener with this name)
         *
```